open_feature/open_feature_client.py

```diff
@@ -260,11 +318,23 @@ def create_provider_evaluation(
         get_details_callable = {
             FlagType.BOOLEAN: self.provider.get_boolean_details,
             FlagType.NUMBER: self.provider.get_number_details,
+            FlagType.INTEGER: self.provider.get_number_details,
+            FlagType.FLOAT: self.provider.get_number_details,
             FlagType.OBJECT: self.provider.get_object_details,
             FlagType.STRING: self.provider.get_string_details,
         }.get(flag_type)
 
         if not get_details_callable:
             raise GeneralError(error_message="Unknown flag type")
 
-        return get_details_callable(*args)
+        value = get_details_callable(*args)
+
+        converter = {
+            FlagType.FLOAT: float,
+            FlagType.INTEGER: int,
+        }.get(flag_type)
+
+        if converter:
+            value.value = converter(value.value)
```

Comment:
This should probably check if the type is a mismatch. If you're expecting a float and get a string, this would be where you get the `ValueError`.

Reply:
Additionally, if you're asking for an integer and get `3.14`, is this a failure state? You didn't get the data type you had expected.